File: aurea.m
% Cálculo da razão áurea (phi)

phi = 3;            % chute inicial
iter = uint8(0);    % contador de iterações (0–255)

while phi ~= sqrt(phi + 1)
    phi = sqrt(phi + 1);
    iter = iter + 1;
end

str = [num2str(iter,'%u'),' iterações: ϕ = ',num2str(phi,'%1.15f')];
disp(str)

File: aurea.py
# Cálculo da razão áurea (phi)
import math

phi = 3.0     # chute inicial
tmp = -1.0    # variável temporária
iteration = 0 # contador de iterações

while (abs(phi - tmp) > 10**(-15)):
    tmp = phi
    phi = math.sqrt(phi + 1)
    iteration += 1

print(iteration, "iterações: ϕ = ", phi)
